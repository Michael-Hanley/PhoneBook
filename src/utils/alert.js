import Alert from '@/components/Alert';
import Vue from 'vue';

const alert = (ref, message, onOk, onClose) => {
  let ComponentClass = Vue.extend(Alert);
  let instance = new ComponentClass();

  instance.onOk = onOk;
  instance.onClose = onClose;
  instance.$slots.default = [message];

  instance.$mount();
  ref.appendChild(instance.$el);
}

export default alert;