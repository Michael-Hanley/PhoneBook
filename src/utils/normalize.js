export const normalizePhone = (value) => {
  if (!value) { return value; }

  const nums = value.replace(/[^\d]/g, '');
  // ex: 12
  if (nums.length <= 3) { return nums; }
  // ex: 123-4
  if (nums.length <= 6) { return nums.replace(/^(.{3})(.{0,3})/, '$1-$2'); }
  // ex: 123-456-78
  return nums.replace(/^(.{3})(.{3})(.{0,4}).*/, '$1-$2-$3');
};