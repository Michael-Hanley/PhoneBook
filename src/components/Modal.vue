<template>
  <div id="modalBackdrop" class="modal-backdrop" @click="onClose">
    <Card class="modal">
      <slot></slot>
    </Card>
  </div>
</template>

<script>
import Card from './Card';
import { escape } from '@/utils/escape';

export default {
  name: 'modal',
  mixins: [escape],
  components: {
    Card,
  },
  methods: {
    escapeHandler: function () {
      this.$emit('onClose');
    },
    onClose: function (e) {
      if (e.target.id === 'modalBackdrop') {
        this.$emit('onClose');
      }
    },
  },
};
</script>

<style scoped>
.modal-backdrop {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(0, 0, 0, 0.3);
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 1000;
}
.modal {
  background: #ffffff;
  overflow-x: auto;
  display: flex;
  flex-direction: column;
}
</style>
