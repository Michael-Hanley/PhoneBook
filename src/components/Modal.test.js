import { mount, shallowMount } from '@vue/test-utils';
import Modal from './Modal.vue';
import { escape } from "@/utils/escape";

let wrapper;
let options = {
  slots: {
    default: 'test'
  },
  propsData: {
    onClose: jest.fn()
  },
  mixins: [escape]
}
describe('Modal', () => {
  beforeEach(() => {
    wrapper = shallowMount(Modal, { ...options })
  });
  it('should fire close event on backdrop click', () => {
    wrapper.find('#modalBackdrop').trigger('click');
    expect(wrapper.emitted('onClose')).toHaveLength(1);
  })
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});