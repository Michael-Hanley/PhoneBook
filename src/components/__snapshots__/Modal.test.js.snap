// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly 1`] = `
<div id="modalBackdrop" class="modal-backdrop" onclose="function mockConstructor() {
        return fn.apply(this, arguments);
      }">
  <card-stub class="modal">test</card-stub>
</div>
`;
